src/bot.ts

```diff
@@ -63,23 +62,24 @@ const run = async () => {
 
   deleted = false
 
-  // INFO: if you don't need lastfm info, comment this bit of code
-  /// lastfm start
-  const currentScrobblingTrackData = await lastfm.call<RecentTracks>('user.getRecentTracks', {
-    user: process.env.LASTFM_USERNAME,
-    limit: 1
-  })
+  let scrobbles: number | undefined = 0
 
-  const currentScrobblingTrack = currentScrobblingTrackData.recenttracks.track[0]
+  if (isLastFmUsing) {
+    const currentScrobblingTrackData = await lastfm.call<RecentTracks>('user.getRecentTracks', {
+      user: process.env.LASTFM_USERNAME,
+      limit: 1
+    })
 
-  const scrobblesData = await lastfm.call<TrackInfo>('track.getInfo', {
-    artist: currentScrobblingTrack.artist['#text'],
-    track: currentScrobblingTrack.name,
-    username: process.env.LASTFM_USERNAME
-  })
+    const currentScrobblingTrack = currentScrobblingTrackData.recenttracks.track[0]
 
-  const scrobbles = Number.parseInt(scrobblesData.track?.userplaycount) || undefined
-  /// lastfm end
+    const scrobblesData = await lastfm.call<TrackInfo>('track.getInfo', {
+      artist: currentScrobblingTrack.artist['#text'],
+      track: currentScrobblingTrack.name,
+      username: process.env.LASTFM_USERNAME
+    })
+
+    scrobbles = Number.parseInt(scrobblesData.track?.userplaycount) || undefined
```

Comment:
yeah, seems like `undefined` scenario is indeed possible. i'd consider changing my code a little bit then: instead of using `|| undefined` i'd change it to `?? 0`